src/MongoDB.Bson/Serialization/Serializers/ObjectSerializer.cs

```diff
@@ -92,7 +92,23 @@ public override object Deserialize(BsonDeserializationContext context, BsonDeser
                     var subType = binaryData.SubType;
                     if (subType == BsonBinarySubType.UuidStandard || subType == BsonBinarySubType.UuidLegacy)
                     {
-                        return binaryData.ToGuid();
+#pragma warning disable 618
```

Comment:
The `ObjectSerializer` is a bit of a weird beast because it can't be configured in any way.

I've done the best I can here.